python/katana/lonestar/analytics/deg_count.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np 
+import pyarrow
+
+from katana.property_graph import PropertyGraph
+from katana.datastructures import LargeArray, AllocationPolicy
+from katana.loops import do_all, do_all_operator
+from katana.atomic import (
+    atomic_add,
+)
+
+@do_all_operator()
+def initialize_in_deg(nin, nid):
+    nin[nid] = 0
+
+@do_all_operator()
+def count_in_and_out_deg(graph: PropertyGraph, nout, nin, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        outdeg += 1
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, 1)
+    nout[nid] = outdeg
+
+@do_all_operator()
+def count_weighted_in_and_out_deg(graph: PropertyGraph, nout, nin, weightArr, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        weight = weightArr[edge]
+        outdeg += weight
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, weight)
+    nout[nid] = outdeg
+
+""" 
+   Calculate the (potentially weighted) in & out degrees of a graph
+   Will add the in & out degrees as 2 new node properties of the graph
+   weight_property: an edge property to use in calculating the weighted degree
+   inProp: the property name for the in degree
+   outProp: the property name for the out degree
+"""
+def calculate_degree(graph: PropertyGraph, weight_propery=None, inProp="in_degree", outProp="out_degree"):
+    num_nodes = graph.num_nodes()
+    nout = LargeArray[np.uint64](num_nodes, AllocationPolicy.INTERLEAVED)
+    nin = LargeArray[np.uint64](num_nodes, AllocationPolicy.INTERLEAVED)
+
+    do_all(
+        range(num_nodes),
+        initialize_in_deg(nin.as_numpy()),
+        steal=False #is this ok?
+    )
+
+    # are we calculating weighted degree?
+    if weight_propery == None:
+        do_all(
+            range(num_nodes),
+            count_in_and_out_deg(graph, nout.as_numpy(), nin.as_numpy()),
+            steal=True
+        )
+    else:
+        do_all(
+            range(num_nodes),
+            count_weighted_in_and_out_deg(graph, nout.as_numpy(), nin.as_numpy(), graph.get_edge_property(weight_propery)),
+            steal=True
+        )
```

Comment:
Nit: Instead of writing the loop twice, store the operator in a variable.

```python
if weight_propery == None:
    calculate_operator = count_in_and_out_deg(graph, nout.as_numpy(), nin.as_numpy())
else:
    calculate_operator = count_weighted_in_and_out_deg(graph, nout.as_numpy(), nin.as_numpy(), graph.get_edge_property(weight_propery))
do_all(
        range(num_nodes),
        calculate_operator,
        steal=True
    )
```